storage/engine.go

```diff
@@ -531,6 +531,11 @@ func (e *Engine) DeleteBucket(orgID, bucketID platform.ID) error {
 
 // DeleteBucketRange deletes an entire bucket from the storage engine.
 func (e *Engine) DeleteBucketRange(orgID, bucketID platform.ID, min, max int64) error {
+	// Snapshot to clear the cache to reduce write contention.
```

Comment:
This isn't the right place to snapshot the cache. We may have thousands of buckets, and we don't want to check if we can snapshot the cache thousands of times. Instead, the call to snapshot the cache should probably come from within the retention enforcer.